src/components/frame/v5/pages/MembersPage/utils.ts

```diff
@@ -1,86 +1,110 @@
-import { ColonyRole, Id } from '@colony/colony-js';
+import { Id } from '@colony/colony-js';
 
-import { getRole } from '~constants/permissions.ts';
+import {
+  getInheritedPermissions,
+  getRole,
+  type UserRoleMeta,
+} from '~constants/permissions.ts';
 import { type ColonyContributorFragment, type ColonyFragment } from '~gql';
-import { getAllUserRoles, getUserRolesForDomain } from '~transformers/index.ts';
+import { getUserRolesForDomain } from '~transformers/index.ts';
 import { extractColonyRoles } from '~utils/colonyRoles.ts';
 
 import { type MemberItem } from './types.ts';
 
+interface RoleInfo {
+  role: UserRoleMeta | undefined;
+  isInherited: boolean;
+}
+
+interface GetRoleInfoParams {
+  colonyRoles: ReturnType<typeof extractColonyRoles>;
+  contributorAddress: string;
+  selectedTeamId: number | undefined;
+  isRootDomain: boolean;
+  isMultiSig: boolean;
+}
+
+const getRoleInfo = ({
+  colonyRoles,
+  contributorAddress,
+  selectedTeamId,
+  isRootDomain,
+  isMultiSig,
+}: GetRoleInfoParams): RoleInfo => {
+  const currentTeamPermissions = getUserRolesForDomain({
+    colonyRoles,
+    userAddress: contributorAddress,
+    domainId: selectedTeamId || Id.RootDomain,
+    excludeInherited: true,
+    isMultiSig,
+  });
+
+  const parentPermissions = getUserRolesForDomain({
+    colonyRoles,
+    userAddress: contributorAddress,
+    domainId: Id.RootDomain,
+    isMultiSig,
+  });
+
+  const inheritedPermissions = getInheritedPermissions({
+    parentPermissions,
+    currentPermissions: currentTeamPermissions,
+    isRootDomain,
+  });
+
+  // inheritedPermissions contains only a difference between parrent and current permissions
```

Comment:
Super nitpick! It should be "...between parent and..."
Thank you for this comment, much easier to digest 👌